ehcache-api/src/main/java/org/ehcache/spi/loaderwriter/CacheLoaderWriter.java

```diff
@@ -113,8 +115,18 @@ default Map<K, V> loadAll(Iterable<? extends K> keys) throws BulkCacheLoadingExc
    * @throws Exception in case no values could be written
    */
   default void writeAll(Iterable<? extends Map.Entry<? extends K, ? extends V>> entries) throws BulkCacheWritingException, Exception {
+    Set<K> successes = new HashSet<>();
```

Comment:
There's nothing wrong with this part of the changeset... but I don't think it belongs in this PR. Better to leave this to another time, all this does is confuse the purpose here (and its presence might be causing you to miss other issues with the real code).

Reply:
I still think this change should be considered separately from this PR.